lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -19,25 +22,34 @@ def status
     end
 
     def deploy_succeeded?
-      success = observed_generation == current_generation &&
-        desired_replicas == status_data['readyReplicas'].to_i &&
-        status_data['currentRevision'] == status_data['updateRevision']
+      success = observed_generation == current_generation
+
+      @pods.each do |pod|
+        success &= pod.definition["metadata"]["labels"]["controller-revision-hash"] == status_data['updateRevision']
```

Comment:
This is a new check we're introducing that works for both updateStrategy types because a Pod's `controller-revision-hash` should update and match its StatefulSet's `updateRevision` value once the Pod is created (as opposed to waiting for the Pod to be ready)

Reply:
> I am guessing the pod definitions are updated each time the method is called?

Hey Dhruv, to clarify, are you asking to confirm if this method updates the pod definitions?

Reply:
Yep, I believe so -- @stevenhope found that pod definitions should be updated each time `deploy_succeeded?` is called due to the [resource_watcher](https://github.com/Shopify/krane/blob/master/lib/krane/resource_watcher.rb#L34) running sync at an interval until it succeeds.